src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -128,5 +199,25 @@ public boolean canBeManagedBy(Auth0UserProfile user) {
         return super.canBeManagedBy(user);
     }
 
+    public JourneyState retrieveJourneyState() {
+        JourneyState journeyState = Persistence.journeyStates.getOneFiltered(eq("monitoredTripId", this.id));
+        if (journeyState == null) journeyState = new JourneyState();
+        return journeyState;
+    }
+
+    public Itinerary latestItinerary() {
+        JourneyState journeyState = retrieveJourneyState();
+        if (journeyState.responses.size() > 0) {
+            return journeyState.responses.get(0).plan.itineraries.get(0);
+        }
+        return itinerary;
+    }
+
+    public void addResponse(Response response) {
+        JourneyState journeyState = retrieveJourneyState();
+        journeyState.lastChecked = System.currentTimeMillis();
+        journeyState.responses.add(0, response);
```

Comment:
Seems like this will endlessly add OTP responses to this list. I don't see anywhere else that removes from this list.

Reply:
I'll add some code for the CheckMonitoredTrip logic to clear these out once the trip has ended.